python/tests/unit/arcticdb/version_store/test_api.py

```diff
@@ -89,3 +91,38 @@ def test_get_num_rows_pickled(lmdb_version_store):
     symbol = "test_get_num_rows_pickled"
     lmdb_version_store.write(symbol, 1)
     assert lmdb_version_store.get_num_rows(symbol) is None
+
+
+@pytest.mark.parametrize(
+    "input_mode, expected_append",
+    [
+        ("WRITE", False),
+        ("APPEND", True),
+        (None, False),
+        (StagedDataFinalizeMethod.APPEND, True),
+        (StagedDataFinalizeMethod.WRITE, False),
+        ([], None),
+        {"write", None},
+    ],
+)
+def test_finalize_staged_data(arctic_library_lmdb, input_mode, expected_append):
```

Comment:
This is a good unit test but I think we should also have some integration tests that check we are actually doing the right thing.

The `expected_append==None` case should probably be split to its own test, rather than parameters in this one.